src/core/StelFileMgr.cpp

```diff
@@ -515,7 +524,7 @@ void StelFileMgr::makeSureDirExistsAndIsWritable(const QString& dirFullPath)
 	{
 		// The modules directory doesn't exist, lets create it.
 		qDebug() << "Creating directory " << QDir::toNativeSeparators(uDir.filePath());
-		if (!QDir("/").mkpath(uDir.filePath()))
+		if (!QDir("/").mkpath(uDir.absoluteFilePath()))
```

Comment:
Why are we even using `/` as the origin? If we had simply constructed `QDir` as `QDir{}`, it'd use current directory and handle this case automagically.

Reply:
I had hoped QDir had a static method mkpath(), but no. So I just copied what was there. The important change here is the absoluteFilePath(). Maybe QDir().filePath() would have been another solution.

Reply:
> The important change here is the absoluteFilePath()

This is the point. You introduced a bunch of code just to work around a mistake that consists of three chars. If you simply remove the `"/"` instead of this commit, I'm pretty sure things will work as expected, since `QDir::absoluteFilePath()` also works relative to CWD.

Reply:
I exchanged one method that was not enough by another which works. The other changes attempt to catch further problems. 

I had just understood the QDir("/") as basically a dummy dir object, and the absoluteFilePath() as the solution that fixed the reported issue.   If it really matters, I can alternatively exchange one constructor by the default constructor, change back the other method call, and perform a sequence of tests if that also works. But I can test this only in early July, I am off for a conference.

Reply:
If I create an empty directory and supply a --user-dir
relative to the directory path relative to the directory
in which I issue the "stellarium" command, stellarium
uses it, partially popullat4es it, then fails attempting
to create additional content.

I consider that dangerous because it appears unpredictable
where Stellarium tries to create content if user-directory is
not absolute

It would be better if Stellarium simply failed on a relative path.

Even better if it used realpath() or an equivalent utility to qualify
a relative path relative to the starting directory and used the
resulting value consistently thereafter.

Thanks,
gil

Reply:
This change is supposed to create and/or use a directory even if only a relative name is given, and configure UserDir with an absolute path, avoiding further issues. Have you tested this branch and does it still misbehave on your system?

The comment left by the original developer of this path check implies that in case of missing subdir, it should be created.

Reply:
I like "configure UserDir with an absolute path, avoiding further issues."

Is Linux likewise affected?

Thanks,
gil